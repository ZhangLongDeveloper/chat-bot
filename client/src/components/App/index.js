import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import {
  INIT, HOME, EXPERIENCE,
  EDUCATION, SKILLS, PROJECTS,
  FEEDBACK,
} from '../../constants/routes';

import Disclaimer from '../Disclaimer';
import Home from '../Home';
import Education from '../Education';
import Experience from '../Experience';
import Skills from '../Skills';
import Projects from '../Projects';
import Feedback from '../Feedback';

import Layout from '../../hoc/layout';

const App = () => (
  <BrowserRouter>
    <Switch>
      <Route path={INIT} exact component={Disclaimer} />
      <Layout>
        <Route path={HOME} exact component={Home} />
        <Route path={EXPERIENCE} exact component={Experience} />
        <Route path={EDUCATION} exact component={Education} />
        <Route path={SKILLS} exact component={Skills} />
        <Route path={PROJECTS} exact component={Projects} />
        <Route path={FEEDBACK} exact component={Feedback} />
      </Layout>
    </Switch>
  </BrowserRouter>
);

export default App;
