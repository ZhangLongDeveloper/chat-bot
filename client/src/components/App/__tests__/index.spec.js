import React from 'react';
import { create } from 'react-test-renderer';

import App from '../../App';

jest.mock('../../Pages/Disclaimer', () => 'Disclaimer');
jest.mock('../../Pages/Home', () => 'Home');
jest.mock('../../Pages/Education', () => 'Education');
jest.mock('../../Pages/Experience', () => 'Experience');
jest.mock('../../Pages/Projects', () => 'Projects');
jest.mock('../../Pages/Feedback', () => 'Feedback');
jest.mock('../../../hoc/layout', () => 'Layout');
jest.mock('react-router-dom', () => ({
  Switch: 'Switch',
  Route: 'Route',
  BrowserRouter: 'BrowserRouter',
}));

const mockComponent = props => (
  <App {...props} />
);

describe('components/<App />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
