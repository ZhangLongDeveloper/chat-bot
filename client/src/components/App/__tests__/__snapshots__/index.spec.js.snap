// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/<App /> should render component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component="Disclaimer"
      exact={true}
      path="/"
    />
    <Layout>
      <Route
        component="Home"
        exact={true}
        path="/home"
      />
      <Route
        component="Experience"
        exact={true}
        path="/experience"
      />
      <Route
        component="Education"
        exact={true}
        path="/education"
      />
      <Route
        component="Skills"
        exact={true}
        path="/skills"
      />
      <Route
        component="Projects"
        exact={true}
        path="/projects"
      />
      <Route
        component="Feedback"
        exact={true}
        path="/feedback"
      />
    </Layout>
  </Switch>
</BrowserRouter>
`;
